puzzles/models.py

```diff
@@ -169,6 +173,14 @@ def clear_answer(self, guess):
     def is_solved(self):
         return self.status == Puzzle.SOLVED
 
+    def solved_time(self):
+        if not self.is_solved():
+            return None
+        solved_times = [
+            answer.created_on for answer in self.guesses.filter(status=Answer.CORRECT)
+        ]
+        return max(solved_times, default=None)
```

Comment:
You should do this in the ORM: https://docs.djangoproject.com/en/dev/ref/models/querysets/#latest